markdownlint/TOP005_blanksAroundMultilineHtmlTags/TOP005_blanksAroundMultilineHtmlTags.js

```diff
@@ -0,0 +1,76 @@
+module.exports = {
+  names: ["TOP005", "blanks-around-multiline-html-tags"],
+  description: "Multiline HTML tags should be surrounded by blank lines or code block delimiters",
+  tags: ["html", "blanks"],
+  information: new URL(
+    "https://github.com/TheOdinProject/curriculum/blob/main/markdownlint/docs/TOP005.md"
+  ),
+  function: function TOP005(params, onError) {
+    /**
+     * HTML code in HTML/JSX code blocks should not be flagged.
+     * We only want to flag HTML tags we use for actual markup,
+     * or md code block examples of such.
+     */
+    const IGNORED_FENCE_TYPES = ["html", "jsx"];
+    const ignoredFencesLineNumbers = params.tokens
+      .filter((token) => {
+        return token.type === "fence" && IGNORED_FENCE_TYPES.includes(token.info);
+      })
+      .map((token) => [token.map[0], token.map[1] - 1]); // 0-indexing
```

Comment:
Could you go over why we're subtracting 1 from the second map item, but not adding 1 to the first map item? If I'm understanding, this returns the range of the MD file for the fenced code blocks, which ends up returning the line just before the opening fence (usually a blank line) and the lline just before the closing fence. So:

````
// blank line
```html
<div>
stuff
</div>
```
````

Assuming this is the start of a file, will have a map of `[1, 5]`, or from `// blank line` to `</div>`.

Could we instead just capture either all of the fence (opening and closing) or just the content inside the fence, and adjust the following helper accordingly?

It looks like this works so probably more of a major nit that just happens to bug me slightly that I honestly wouldn't block over, or I may be not understanding what else this may be doing 😆

Reply:
A Token's `map` property is an array of 2 ints:
1. params.lines index for the opening line
2. params.lines index for the line **after** the closing line

So `token.map[0]` is the params.lines index for the opening triple backticks, and `token.map[1]` is the params.lines index of the line after the closing triple backticks, which will be a blank line in valid files.

While it'd technically still work if I don't `- 1` from `token.map[1]`, doing so means I now have the params.lines indices for the opening and closing backticks themselves, which I felt more semantically correct. Not subtracting 1 will actually make no difference when calling `isWithinIgnoredFence` so if omitting it significantly improves clarity, I'll gladly do so.

Reply:
Since returning just `token.map` as is will have absolutely no effect on the outcome of the `isWithinIgnoredFence` check (since all it'll do is include the closing backticks line in the range), I've opted to go for that to reduce the number of "unintuitive" actions that would require comments.